### DISCUSSION TITLE
The MessageStatus is always Removed [IT is NOT a Bug]

### DISCUSSION THREAD

**Discussion by ghost** [jchristn/WatsonTcp#139] [General]
Hello! It is amazing this library.

But, I had a problem with the reason parameter for the disconnection. When I pass MessageStatus different from 'Removed' to 'DisconnectClient' method, my delegate Disconnect method is called and the e.reason ever is MessageStatus.Removed. I tested directly in WatsonTcp Code and noticed that is a bug.
I didn't find a solution yet, I realized the bug now.

Thanks a lot!!

**Comment by jchristn** [comment]
Hi @StephesonAlves thanks for your kind words - what version are you using?  I just tested on the latest (4.8.5) after amending the ```Test.Server``` project (to be able to pass in the reason) and it seems to work.  

Here's an example of calling it with ```MessageStatus.Shutdown```.

![image](https://user-images.githubusercontent.com/1421081/119013992-62bae280-b94c-11eb-845f-514138a6736b.png)

Here is the commit that includes the change to ```Test.Server```: https://github.com/jchristn/WatsonTcp/commit/60e3c88631556934d311f3916e8cd3a8bde9937c

**Reply by ghost** [reply]
I'm sorry! 
Now I understand the situation.
In truth, I tried to do a kick the client from the Watson server with a specific reason and needed to catch this disconnection reason of the client yet in the server disconnect method by 'DisconnectionEventArgs.reason' to write in the server console log. I saw that this disconnection reason from the server is sent to the client and isn't the same in the server. I thought it was, but now I understand the real reason. It makes sense!!
 **IT is NOT a Bug!** 👍 

I solved my problem by creating a property that saves the client status.


![watsonTCP](https://user-images.githubusercontent.com/49839610/119057611-26ff3780-b9a3-11eb-88bc-ec81b0ae0294.png)



Thanks a lot, man!

**Reply by jchristn** [reply]
Glad it's working for you!  Cheers, Joel
